File: Programming Basics with JS/T07 - Nested Loops/exercise/05. Special Numbers.js
function specialNumbers(input) {
    let n = Number(input[0]);
    let printLine = '';

    for (let i = 1; i <= 9; i++) {
        for (let j = 1; j <= 9; j++) {
            for (let k = 1; k <= 9; k++) {
                for (let l = 1; l <= 9; l++) {
                    if (n % i === 0 && n % j === 0 && n % k === 0 && n % l === 0) {
                        printLine += `${i}${j}${k}${l} `;
                    }
                }
            }
        }
    }
    console.log(printLine);
}
specialNumbers(['3']);

File: Programming Basics with Python/T07 - Nested Loops/exercise/05. Special Numbers.py
number = int(input())

for current in range(1111, 10000):
    number_is_special = True
    current_as_string = str(current)

    for current_digit in current_as_string:
        if int(current_digit) == 0 or number % int(current_digit) != 0:
            number_is_special = False
            break
    if number_is_special:
        print(current, end=" ")